federationapi/routing/routing.go

```diff
@@ -460,4 +460,10 @@ func Setup(
 			return QueryDeviceKeys(httpReq, request, keyAPI, cfg.Matrix.ServerName)
 		},
 	)).Methods(http.MethodPost)
+
+	v1fedmux.Handle("/openid/userinfo",
+		httputil.MakeExternalAPI("federation_openid_userinfo", func(req *http.Request) util.JSONResponse {
```

Comment:
Is an external API correct here or should I be checking client authorization?

Reply:
Nope, external API sounds right as it's an unauthenticated endpoint.